src/storage/txn/flow_controller.rs

```diff
@@ -0,0 +1,1055 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::collections::VecDeque;
+use std::f64::INFINITY;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::mpsc::{self, Receiver, RecvTimeoutError, SyncSender};
+use std::sync::Arc;
+use std::thread::{Builder, JoinHandle};
+use std::time::Duration;
+use std::u64;
+
+use collections::HashMap;
+use engine_rocks::FlowInfo;
+use engine_traits::KvEngine;
+use rand::Rng;
+use tikv_util::time::{duration_to_sec, Consume, Instant, Limiter};
+
+use crate::storage::config::FlowControlConfig;
+use crate::storage::metrics::*;
+
+const SPARE_TICK_DURATION: Duration = Duration::from_millis(1000);
+const SPARE_TICKS_THRESHOLD: u64 = 10;
+const RATIO_SCALE_FACTOR: f64 = 10000000.0;
+const EMA_FACTOR: f64 = 0.6; // EMA stands for Exponential Moving Average
+const LIMIT_UP_PERCENT: f64 = 0.04; // 4%
+const LIMIT_DOWN_PERCENT: f64 = 0.02; // 2%
+const MIN_THROTTLE_SPEED: f64 = 16.0 * 1024.0; // 16KB
+const MAX_THROTTLE_SPEED: f64 = 200.0 * 1024.0 * 1024.0; // 200MB
```

Comment:
Not sure if this is discussed before, but 200MB seems pretty small. Will the limit be turned on and off repeatedly on fast disk?

Reply:
Personally speaking, not a bad idea for a module introduction.

Reply:
The scheduler writing flow is 100MB, and I haven't seen any disk can reach the 200MB/s for foreground write flow of leader